Python/optimizer.c

```diff
@@ -169,6 +169,7 @@ _PyOptimizer_BackEdge(_PyInterpreterFrame *frame, _Py_CODEUNIT *src, _Py_CODEUNI
     if (err <= 0) {
         assert(executor == NULL);
         if (err < 0) {
+            _PyFrame_SetStackPointer(frame, stack_pointer);
```

Comment:
Oh, interesting. It seems this is needed because when `JUMP_BACKWARD` calls `_PyOptimizer_BackEdge` it doesn't update the stack pointer in the frame, it just passes the stack pointer to the optimizer, which upon success passes it to the executor, which will eventually call `STORE_SP()` to update the stack pointer in the frame. And sure, this PR seems to fix the hole, but I wonder if it wouldn't be more robust to just save it before even calling `_PyOptimizer_BackEdge`.